.github/workflows/codespell.yml

```diff
@@ -0,0 +1,25 @@
+# Codespell configuration is within .codespellrc
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Annotate locations with typos
+        uses: codespell-project/codespell-problem-matcher@v1
```

Comment:
I'm a bit nervous about running actions like this from other projects, purely because if that project gets compromised my own CI runs could end up running untrusted code.

My preferred pattern for codespell is to run it like this: https://github.com/simonw/datasette/blob/main/.github/workflows/spellcheck.yml

Reply:
I hear you. FWIW

```
permissions:
  contents: read
```

above should at least make it unable to push... as as secrets are also not exposed implicitly, should be safe?? (I am really curious now too ;) )